### PR TITLE
search analtics relateed changes

### PR DESCRIPTION
**Pull request by arpit-twnscrpt**

### CODE REVIEW

**Review by banoresaurabhTS — APPROVED** [review]

File: src/model/search-stream-helper-model/event-data.model.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { float } from "aws-sdk/clients/lightsail";
+
+export class EventData {
+    event_id: number | undefined;
```
Comment (pratik-townscript) [inline src/model/search-stream-helper-model/event-data.model.ts]: event_id?: number could solve this same issue. Is there any additional benefits of using undefined with pipe ("|") here? 
Rename to EventDataModel

File: src/service/collector/search-data-collector.service.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import {SearchStreamData} from "../../model/search-stream.model";
+import {RecordSearchData} from "../publisher/record-search-data.service";
+import {UUID} from "../shared/UUID";
+import { EventData } from "../../model/search-stream-helper-model/event-data.model";
+import { OrganizerData } from "../../model/search-stream-helper-model/organizer-data.model";
+import { SearchSuggestionData } from "../../model/search-stream-helper-model/search-suggestion-data.mode";
+import { UserMetaDataService } from "../shared/user-metadata.service";
+declare var window: any;
+
+export class SearchDataCollector {
```
Comment (pratik-townscript) [inline src/service/collector/search-data-collector.service.ts]: Rename it to SearchDataCollectorService or remove the file as service

File: src/model/search-stream-helper-model/organizer-data.model.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export class OrganizerData {
```
Comment (pratik-townscript) [inline src/model/search-stream-helper-model/organizer-data.model.ts]: Rename OrganizerDataModel

File: src/model/search-stream-helper-model/search-suggestion-data.mode.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import { float } from "aws-sdk/clients/lightsail";
+
+export class SearchSuggestionData {
+    suggestion: String | undefined;
```
Comment (pratik-townscript) [inline src/model/search-stream-helper-model/search-suggestion-data.mode.ts]: File name is "mode". I guess you wanted to use model. Also rename the class

File: src/model/search-stream.model.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import { EventData } from "./search-stream-helper-model/event-data.model";
+import { OrganizerData } from "./search-stream-helper-model/organizer-data.model";
+import { SearchSuggestionData } from "./search-stream-helper-model/search-suggestion-data.mode";
+
+export interface SearchStreamData {
```
Comment (pratik-townscript) [inline src/model/search-stream.model.ts]: Renam

File: src/model/StreamData.ts [hunk 6]
```diff
@@ -1,24 +1,11 @@
-export interface StreamData {
+import { BasicStreamDataModel } from "./basic-stream-data.model";
+
+export interface StreamData extends BasicStreamDataModel {
```
Comment (arpit-twnscrpt) [inline src/model/StreamData.ts]: EventStreamDataModel name

File: src/model/search-stream.model.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import { EventDataModel } from "./search-stream-helper-model/event-data.model";
+import { OrganizerDataModel } from "./search-stream-helper-model/organizer-data.model";
+import { SearchSuggestionDataModel } from "./search-stream-helper-model/search-suggestion-data.model";
+import { BasicStreamDataModel } from "./basic-stream-data.model";
+
+export interface SearchStreamDataModel extends BasicStreamDataModel {
+    typedText: string, //original text typed in
```
Comment (arpit-twnscrpt) [inline src/model/search-stream.model.ts]: add ? for not-neccesary fields

File: src/service/publisher/record-search-data.service.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
+import {SearchStreamDataModel} from "../../model/search-stream.model";
+import * as Kinesis from "aws-sdk/clients/kinesis";
+import {SearchDataCollectorService} from "../collector/search-data-collector.service";
+import {Configuration} from "../../model/Configuration";
+
+export class RecordSearchData {
```
Comment (arpit-twnscrpt) [inline src/service/publisher/record-search-data.service.ts]: Combine both record data services

File: src/service/producer/search-data-producer.service.ts [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+        SearchDataCollectorService.configure(config);
+        if(disable){
+            SearchDataCollectorService.disable();
+        }
+    };
+
+    static callSearchDataCollector = (typedText:string, searchIntent:string, searchSuggestions:[], eventCount:number,
+                                eventsListed:[], organizersListed:[], loggedInUserId: string, pageNumber: number)=>{
+        try{
+            let relativeUrl = window.location.pathname;
```
Comment (arpit-twnscrpt) [inline src/service/producer/search-data-producer.service.ts]: Reuse common code for both producers

**Review by pratik-townscript — APPROVED** [review]

**Pull request merged**